tests/pytests/test_cursors.py

```diff
@@ -416,3 +416,43 @@ def testCursorDepletionStrictTimeoutPolicy():
     env.expect(
         'FT.AGGREGATE', 'idx', '*', 'LOAD', '1', '@t', 'GROUPBY', '1', '@t', 'WITHCURSOR', 'COUNT', str(num_docs), 'TIMEOUT', '1'
     ).error().contains('Timeout limit was reached')
+
+@skip(cluster=True)
+def test_mod_6597(env):
+    """Tests that we update the numeric index appropriately upon deleting
+    documents from a numeric index, and are able to query an invalid cursor in
+    such case getting an empty result instead of a crash."""
+    conn = getConnectionByEnv(env)
+
+    # Create an index with a numeric field.
+    env.expect('FT.CREATE', 'idx', 'ON', 'HASH', 'SCHEMA', 'test', 'NUMERIC').equal('OK')
+
+    # Populate the db (and index) with corresponding documents (one more than
+    # the `forkGcCleanThreshold`, which is currently set to 100).
+    num_docs = 101
+    for i in range(num_docs):
+        conn.execute_command('hset', f'doc{i}', 'test', str(i))
+
+    # Initialize a cursor
+    res, cid = env.execute_command('ft.aggregate', 'idx', f'@test:[1 {num_docs}]', 'LOAD', '1', '@test', 'WITHCURSOR', 'COUNT', '1')
+    n = len(res) - 1
+
+    # Make sure GC is not self-invoked (periodic run).
+    env.expect('FT.CONFIG', 'SET', 'FORK_GC_RUN_INTERVAL', 3600).equal('OK')
```

Comment:
Consider using `num_docs` to be less than `FORK_GC_CLEAN_THRESHOLD` so it is stable

Reply:
Not sure I understand your meaning.
If it is set to less than `FORK_GC_CLEAN_THRESHOLD` the GC won't clean the index.

Reply:
Sorry, I meant equals not less than.

Reply:
Added 👍